### PR TITLE
[Tools] Release helper Update

### PR DESCRIPTION
**Pull request by BigCat20196**
https://github.com/Azure/azure-sdk-for-python/issues/21557

### CODE REVIEW

File: scripts/release_helper/common.py [hunk 1]
```diff
@@ -308,6 +314,7 @@ def output(self):
             file_out.write('| issue | author | package | assignee | bot advice | created date of issue | target release date | date from target |\n')
             file_out.write('| ------ | ------ | ------ | ------ | ------ | ------ | ------ | :-----: |\n')
             for item in self.result:
+                print(f'**** item: {item.issue_package.issue.number}')
```
Comment (msyyc) [inline scripts/release_helper/common.py:317]: use `_LOG` instead of `print` to log

File: scripts/release_helper/python.py [hunk 2]
```diff
@@ -1,23 +1,68 @@
...
+        self.get_readme_link(origin_link)
+
+    def get_package_name(self) -> None:
+        self.init_readme_link()
+        pattern_resource_manager = re.compile(r'/specification/([\w-]+/)+resource-manager')
+        readme_python_path = pattern_resource_manager.search(self.readme_link).group() + '/readme.python.md'
+        contents = str(self.issue_package.rest_repo.get_contents(readme_python_path).decoded_content)
+        pattern_package = re.compile(r'package-name: [\w+-.]+')
+        self.package_name = pattern_package.search(contents).group().split(':')[-1].strip()
+        print(f'*** package_name: {self.package_name}')
```
Comment (msyyc) [inline scripts/release_helper/python.py:39]: same with https://github.com/Azure/azure-sdk-for-python/pull/24800/files#r898716202

File: scripts/release_helper/python.py [hunk 3]
```diff
@@ -1,23 +1,68 @@
...
 
 # assignee dict which will be assigned to handle issues
-_PYTHON_OWNER = {'BigCat20196', 'msyyc'}
+_PYTHON_OWNER = {'BigCat20196', 'msyyc', 'azure-sdk'}
 
 # 'github assignee': 'token'
 _ASSIGNEE_TOKEN_PYTHON = {
     'BigCat20196': os.getenv('PYTHON_BIGCAT_TOKEN'),
     'msyyc': os.getenv('PYTHON_MSYYC_TOKEN'),
 }
```
Comment (msyyc) [inline scripts/release_helper/python.py:15]: change to use `azure-sdk` token. Thus, there is no need to use dict and set shall be enough.

File: scripts/release_helper/python.py [hunk 4]
```diff
@@ -1,23 +1,68 @@
...
 
+# record published issues
+_FILE_OUT = 'published_issues_python.csv'
```
Comment (msyyc) [inline scripts/release_helper/python.py:18]: be careful to pollute the file when you debug

File: scripts/release_helper/python.py [hunk 5]
```diff
@@ -1,23 +1,68 @@
...
+            self.issue_package.issue.create_comment(body=comment)
+            self.issue_package.issue.edit(state='closed')
+            self.issue_package.issue.add_to_labels('auto-closed')
+            self.log("has been closed!")
+            record_release(self.package_name, self.issue_package.issue, _FILE_OUT)
+
+    def run(self) -> None:
+        self.get_package_name()
+        super().run()
+        self.auto_close()
```
Comment (msyyc) [inline scripts/release_helper/python.py:59]: we should make sure `auto-closed` issue is not recorded. To do that, we could add `open` attribute in common process. And make it `False` in `auto_close()`.

File: scripts/release_helper/common.py [hunk 6]
```diff
@@ -334,18 +347,20 @@ def output_md(item: IssueProcess):
...
             try:
                 issue.run()
                 self.result.append(issue)
             except Exception as e:
-                _LOG.error(f'Error happened during handling issue {item.issue.number}: {e}')
-        self.output()
+                self.log_error(f'Error happened during handling issue {item.issue.number}: {e}')
 
+    def run(self):
+        self.get_result()
```
Comment (msyyc) [inline scripts/release_helper/common.py:361]: `get_resulte` is a little too general, `proc_issue` is better

File: scripts/release_helper/go.py [hunk 7]
```diff
@@ -3,22 +3,19 @@
 import os
 
 # assignee dict which will be assigned to handle issues
-_GO_OWNER = {'ArcturusZhang', 'Alancere'}
-
-# 'github assignee': 'token'
-_ASSIGNEE_TOKEN_GO = {'Alancere': os.getenv('AZURESDK_BOT_TOKEN')}
+_GO_OWNER = {'ArcturusZhang', 'Alancere', 'azure-sdk'}
```
Comment (msyyc) [inline scripts/release_helper/go.py:6]: not all owners will handle issues. We still need two set:

- `LANGUAGE_ASSIGNEE`: person who handle issue
- `LANGUAGE_ADDITIONAL_OWNER`: person who is not assignee but may comment

File: scripts/release_helper/utils.py [hunk 8]
```diff
@@ -28,6 +38,96 @@ def get_origin_link_and_tag(issue_body_list: List[str]) -> (str, str):
...
+
+    # Create a connection to the org
+    credentials = BasicAuthentication('', personal_access_token)
+    run_parameters = models.RunPipelineParameters(**paramaters)
+    client = PipelinesClient(base_url=organization_url, creds=credentials)
+    result = client.run_pipeline(project='internal', pipeline_id=2500, run_parameters=run_parameters)
+    if result.state == 'inProgress':
+        return True
+    else:
+        return False
```
Comment (msyyc) [inline scripts/release_helper/utils.py:115]:
```suggestion
    return result.state == 'inProgress':
```

File: scripts/release_helper/release_helper.yml [hunk 9]
```diff
@@ -54,10 +54,8 @@ jobs:
 
         # import env variable
         export TOKEN=$(Yuchao-GitToken)
```
Comment (msyyc) [inline scripts/release_helper/release_helper.yml:56]: Could we remove this token, too?

File: scripts/release_helper/release_helper.yml [hunk 10]
```diff
@@ -54,10 +54,8 @@ jobs:
...
-        export PYTHON_BIGCAT_TOKEN=$(Jiefeng-GitToken)
-        export PYTHON_ZED_TOKEN=$(Zed-GitToken)
-        export PYTHON_MSYYC_TOKEN=$(Yuchao-GitToken)
         export AZURESDK_BOT_TOKEN=$(azuresdk-github-pat)
+        export PIPELINE_TOKEN=$(PIPELINE-TOKEN)
```
Comment (msyyc) [inline scripts/release_helper/release_helper.yml:58]: `AZURE_BOT_TOKEN` does not have access to trigger pipeline?

File: scripts/release_helper/utils.py [hunk 11]
```diff
@@ -1,12 +1,22 @@
...
+
+from bs4 import BeautifulSoup
 from github.Issue import Issue
 from github.Repository import Repository
-import logging
-from typing import List
+import requests
+from azure.devops.v6_0.pipelines.pipelines_client import PipelinesClient
+from azure.devops.v6_0.pipelines import models
+from msrest.authentication import BasicAuthentication
```
Comment (msyyc) [inline scripts/release_helper/utils.py:13]: Are these new libs needed in `requirements.txt`?

File: scripts/release_helper/python.py [hunk 12]
```diff
@@ -1,25 +1,156 @@
...
+    def __init__(self, issues, language_owner):
+        super(Python, self).__init__(issues, language_owner)
+        self.file_out_name = 'release_python_status.md'
+        self.issue_process_function = IssueProcessPython
+
+    def duplicated_policy(self):
+        count = Counter([item.package_name for item in self.result])
+        for item in self.result:
+            if count[item.package_name] > 1:
+                item.bot_advice.insert(0, 'duplicated issue  <br>')
```
Comment (msyyc) [inline scripts/release_helper/python.py:146]: `count` => `counter` is better

File: scripts/release_helper/release_helper.yml (hunk 10 above)
Comment (BigCat20196) [reply scripts/release_helper/release_helper.yml:58]: yes, the Azure_BOT_TOKEN is github token, but it needs Azure DevOps token.

Comment (BigCat20196) [reply scripts/release_helper/release_helper.yml:58]: > `AZURE_BOT_TOKEN` does not have access to trigger pipeline?

yes, the Azure_BOT_TOKEN is github token, but it needs Azure DevOps token.

File: scripts/release_helper/release_helper.yml (hunk 9 above)
Comment (BigCat20196) [reply scripts/release_helper/release_helper.yml:56]: > Could we remove this token, too?

yeah

**Review by msyyc — APPROVED** [review]

**Pull request merged**